openedx/features/course_experience/course_tools.py

```diff
@@ -0,0 +1,64 @@
+"""
+Support for course tool plugins.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+# Stevedore extension point namespace
+COURSE_TOOLS_NAMESPACE = 'openedx.course_tool'
+
+
+class CourseTool(object):
+    """
+    This is an optional base class for Course Tool plugins.
+
+    Plugin implementations can choose to subclass CourseTool to get
+    useful default behavior, but it is not a requirement. A course tool
+    plugin must define all of the methods in this class.
+    """
+
+    @classmethod
+    def is_enabled(cls, course_key):
+        """
+        Returns true if this tool is enabled for the specified course key.
+        """
+        raise NotImplementedError("Must specify if tool is enabled.")
```

Comment:
1. I think @andy-armstrong had a default implementation for this that returned true.  That shouldn't have changed.
2. Please don't squash commits until after the review has been approved, so I can more easily see what is changing.  Thanks!